irc/client.go

```diff
@@ -777,17 +778,18 @@ func (client *Client) updateNickMaskNoMutex() {
 	}
 
 	client.nickMaskString = fmt.Sprintf("%s!%s@%s", client.nick, client.username, client.hostname)
-	client.nickMaskCasefolded = fmt.Sprintf("%s!%s@%s", client.nickCasefolded, client.usernameCasefolded, cfhostname)
+	client.nickMaskCasefolded = fmt.Sprintf("%s!%s@%s", client.nickCasefolded, strings.ToLower(client.username), cfhostname)
 }
 
 // AllNickmasks returns all the possible nickmasks for the client.
 func (client *Client) AllNickmasks() (masks []string) {
 	client.stateMutex.RLock()
 	nick := client.nickCasefolded
-	username := client.usernameCasefolded
+	username := client.username
 	rawHostname := client.rawHostname
 	vhost := client.getVHostNoMutex()
 	client.stateMutex.RUnlock()
+	username = strings.ToLower(client.username)
```

Comment:
is it just me or could this `strings.ToLower()` just be done on the initial assignment up on line 788?

Reply:
It could --- this is just me being neurotic about doing unnecessary computation while holding locks :-p

I can move it, NBD

Reply:
I can get that and thought the same, but with this line we're grabbing `client.username` directly after we've released the `client.stateMutex`, I'd be fine with `username = strings.ToLower(username)` as well if that's what was intended